### ISSUE TITLE
lwt opam package > 4.1.0 shows version of "dev"

### ISSUE THREAD

**Issue by cmkarlsson** [ocsigen/lwt#690]
When installing the latest lwt packages the version given by `ocamlfind query` shows version: `dev`

Tested with Ocaml 4.06.1 and 4.07.1.

How to reproduce:
```
$ opam switch create 4.07.1
$ eval $(opam env)
$ opam install lwt.4.2.1
$ opam install ocamlfind
$ ocamlfind query lwt -l |grep version
version:     dev
$ opam install lwt.4.1.0
$ ocamlfind query lwt -l |grep version
version:     4.1.0
```

Having the version as `dev` prevents certain packages from installing correctly (stog).

Looking at the previous releases the lwt.opam file is usually change to show the correct version. Perhaps this was missed in the latest releases?

**Comment by aantron** [comment]
I stopped adding the `version:` field for newer releases, because I vaguely remember an opam maintainer objecting to it, due to redundancy with the opam directory structure. However, that's clearly not fully compatible with Dune's `META` file generation, so I'm adding them back in in ocaml/opam-repository#14043. Thanks.

**Comment by mseri** [comment]
@aantron I think you should reopen this for now, the submitted fix does not seem to be the correct fix

**Comment by kit-ty-kate** [comment]
i agree. This should be reopened

**Comment by aantron** [comment]
@mseri, @kit-ty-kate, thanks for correcting me. So, what should the `version:` field be set in release sources? I understand that `version: the-actual-version` will work, but I also saw the suggestion to leave it out entirely, and it looks like Dune code you linked does something intelligent in that case.

Is it best to simply delete `version:` fields from the `.opam` files in the repo?

**Comment by mseri** [comment]
If you drop the version field altogether, opam will infer it from the environment (you would need to pin it as lwt.dev to make sure you get the dev version installed). I think that that would also be the version used by `dune` if the field is not present.

Otherwise you may need to add a dune profile for the dev environment. I think it would be better to ask one of the dune developers to figure out the best solution.

**Comment by aantron** [comment]
I propose to simply return to my former practice starting with the next release, which is creating a little release branch with the correct version numbers, and tagging that branch. Hopefully, after a while, almost nobody will be installing 4.2.0 and 4.2.1 anymore.

**Comment by aantron** [comment]
Closing this. It will be addressed for Lwt > 4.2.1 by the next release.
